modules/oe_multilingual_url_suffix/src/Event/UrlSuffixesAlterEvent.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_multilingual_url_suffix\Event;
+
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Dispatched by LanguageNegotiationUrlSuffix::getUrlSuffixes().
+ *
+ * Allows to alter the list of url suffixes retrieved from config.
+ */
+class UrlSuffixesAlterEvent extends Event {
+
+  const EVENT = 'oe_multilingual_url_suffix.url_suffixes_alter';
+
+  /**
+   * The list of url suffixes.
+   *
+   * @var array
+   */
+  protected $urlSuffixes;
```

Comment:
Typically we have a constructor for these to intake the initial values to be passed around using the event (the suffixes in this case). So when you instantiate the event later, you can directly pass them without the setter.